web/src/main/java/org/wso2/testgrid/web/operation/PipelineManager.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.testgrid.web.operation;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.hc.client5.http.fluent.Request;
+import org.apache.hc.core5.http.ContentType;
+import org.apache.hc.core5.http.HttpResponse;
+import org.apache.hc.core5.http.HttpStatus;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.wso2.testgrid.web.utils.Constants.BLUE_OCEAN_URI;
+import static org.wso2.testgrid.web.utils.Constants.JENKINS_HOME;
+import static org.wso2.testgrid.web.utils.Constants.JENKINS_USER_AUTH_KEY;
+
+import java.io.IOException;
+
+import static javax.ws.rs.core.HttpHeaders.USER_AGENT;
+
+/**
+ * Implementation of creating pipeline jobs from config.xml files.
+ */
+public class PipelineManager {
+    private static final Logger logger = LoggerFactory.getLogger(PipelineManager.class);
+
+    /**
+     * Creates new pipeline job in Jenkins server by calling its REST API.
+     * @param configXml configuration file for the new job.
+     * @param jobName name for the new job.
+     * @return URL to check the status of the new job.
+     */
+    public String createNewPipelineJob(String configXml, String jobName) throws Exception {
+        HttpResponse response = Request.Post(JENKINS_HOME + "/createItem?name=" + jobName)
```

Comment:
What if jobName was null? I could not find where the input validation has done.

Reply:
If the jobName is null (or any name with jenkin's reserved characters), the Jenkins will send us back the error when trying to create a job. We will be passing that back to user of our API as well. Wouldn't that be sufficient and more accurate as the validation is done by Jenkin itself?